File: app/src/main/java/com/essence/mvvmapplication/posts/domain/PostsServices.java
package com.essence.mvvmapplication.posts.domain;

import com.essence.mvvmapplication.posts.model.PostModel;

import java.util.List;

import io.reactivex.rxjava3.core.Observable;
import retrofit2.http.GET;

public interface PostsServices {
    @GET("posts")
    Observable<List<PostModel>> getPosts();
}

File: app/src/main/java/com/essence/mvvmapplication/posts/domain/PostsServices.kt
package com.essence.mvvmapplication.posts.domain

import retrofit2.http.GET
import com.essence.mvvmapplication.posts.model.PostModel
import io.reactivex.rxjava3.core.Observable

interface PostsServices {
    @get:GET("posts")
    val posts: Observable<List<PostModel?>?>?
}